app/view/sdl/ResetTimeoutPopUp.js

```diff
@@ -0,0 +1,341 @@
+
+    /*
+     * play function. plays the file
+     */
+    play: function (files){
+        if (files != '') {
+            var files_to_play = files.split('\n');
+            for (var i = 0; i < files_to_play.length; ++i) {
+              this.player.addFile(files_to_play[i]);
+            }
+            this.player.playFiles();
+        }
+    },
+    
+     /*
+      * setContext function. sets the text displayed in the pop-up
+      */
+    setContext: function(msg){
+        if('string' == typeof msg) {
+            this.contextView.set('content',msg)
+        }
+    },
+    timeoutSeconds: {},
+    timeoutString:'',
+    defaultTimeout: 10,
+    resetPeriod: 10,
+    timer: null,
+    callbacks: {},
+    resetTimeoutCallback: {},
+    requestIDs: {},
+    resetTimeoutRPCs: [],
+    
+    /*
+     * extendResetTimeoutRPCs function. Appends to current resetTimeoutRPCs 
+     * list a new elements
+     */ 
+    extendResetTimeoutRPCs: function(resetTimeoutRPCs) {
+        if(null == resetTimeoutRPCs){
+            resetTimeoutRPCs = [];
+        }
+        this.resetTimeoutRPCs = this.resetTimeoutRPCs.concat(resetTimeoutRPCs);
+    },
+
+    /**
+     * extendResetTimeoutCallBack function. expand callbacks that will be called when the 
+     * resetTimeout button is pressed
+     */
+    extendResetTimeoutCallBack: function(resetTimeoutCallback, method) {
+        this.resetTimeoutCallback[method] = resetTimeoutCallback;
+    },
+
+    /*
+     * expandCallbacks function. expand callbacks that will be called when the 
+     * respondButton button is pressed
+     */     
+    expandCallbacks: function(callback, method){
+        if('function' == typeof callback) { 
+            this.callbacks[method] = callback;
+        }
+    },
+
+    /**
+     * addCheckBox function for add Check box and label to the popUp view
+     * for reseting the timeouts and
+     * sending the response if RPC more then one 
+     */
+    addCheckBox: function(length) {
+        var list = this.get('contextView.listOfRPC.list.childViews');
+        list.clear();
+        for(var i=0; i<length;++i){
+            list.pushObject(Em.Checkbox.create(
+                {
+                  elementId: this.resetTimeoutRPCs[i] + 'checkBox',
+                  classNames: 'component',
+                  checked: true,
+                  disabled: this.resetTimeoutRPCs[i] == 'UI.PerformInteraction' ? true : false
+                }
+              )),
+
+              list.pushObject(SDL.Label.create({
+                elementId: this.resetTimeoutRPCs[i] + 'Label',
+                classNames: 'component',
+                content: this.resetTimeoutRPCs[i]
+              }))
+        }
+    },
+
+    /**
+     * setDefaultTimeout function for set default timeout
+     */
+    setDefaultTimeout: function()
+    {
+        self = this;
+        this.resetTimeoutRPCs.forEach(function (method) {
+            if (self.timeoutSeconds[method] == undefined) {
```

Comment:
Better to use `in` operator to check that the key is defined. Also check all of them in PR